lib/debug/chrome_debugger.dart

```diff
@@ -0,0 +1,638 @@
+      return Future.forEach(uris, (String uri) {
+        Mapping m = reversedMaps[uri];
+        launch.pipeStdio('Set breakpoint for: $uri\n');
+        if (m == null) return new Future.value();
+
+        SourceMapSpan span = SingleMappingProxy.spanFor(m, atomBreakpoint.line - 1, atomBreakpoint.column ?? 0);
+        if (span == null) return new Future.value();
+
+        return setBreakpointByUrl(span.sourceUrl.toString(), span.start.line, span.start.column)
+            .then((Breakpoint chromeBreakpoint) {
+              launch.pipeStdio('  Set breakpoint: $chromeBreakpoint\n');
+              if (chromeBreakpoint == null) return;
+              breakpoints.putIfAbsent(atomBreakpoint, () => []).add(chromeBreakpoint);
+            }).catchError((e) {
+              launch.pipeStdio('  Fail to set breakpoint: $atomBreakpoint\n');
+            });
+      });
+    }).catchError((e) {
+      launch.pipeStdio(
+          '  Error resolving uri: ${atomBreakpoint.path}\n'
+          '    ${e}\n');
+    });
+  }
+
+  Future removeBreakpoint(AtomBreakpoint atomBreakpoint) {
+    List<Breakpoint> bkpts = breakpoints.remove(atomBreakpoint);
+    if (bkpts != null) {
+      return Future.forEach(bkpts, (Breakpoint breakpoint) {
+        return chrome.debugger.removeBreakpoint(breakpoint.breakpointId)
+            .catchError((e) {
+          launch.pipeStdio('Error removing breakpoint:\n  $e\n');
+        });
+      });
+    }
+    return new Future.value();
+  }
+
+  Future installBreakpoints() {
+    return Future.forEach(breakpointManager.breakpoints, (AtomBreakpoint atomBreakpoint) {
+      if (breakpoints[atomBreakpoint] != null || !atomBreakpoint.fileExists()) {
+        return null;
+      }
+      return addBreakpoint(atomBreakpoint);
+    });
+  }
+
+  Future setBreakpointByUrl(String url, int line, int column) {
+    return chrome.debugger.setBreakpointByUrl(line, url: url, columnNumber: column)
+        .then((bk) {
+      launch.pipeStdio('Breakpoint added: $bk\n');
+      return bk;
+    }).catchError((e) {
+      launch.pipeStdio('Breakpoint not added: $url($line,$column)\n  $e\n');
+    });
+  }
+
+  void dispose() {
+    subs.cancel();
+    if (isAlive) terminate();
+    chrome.close();
+    uriResolver.dispose();
+  }
+
+  bool get isAlive => launch.isRunning;
+
+  Stream<DebugIsolate> get onPaused => _isolatePaused.stream;
+  Stream<DebugIsolate> get onResumed => _isolateResumed.stream;
+
+  Future get onTerminated => completer.future;
+
+  Future resume() => chrome.debugger.resume();
+
+  stepIn() {
+    if (isPaused) {
+      isPaused = false;
+      chrome.debugger.stepInto();
+    }
+  }
+  stepOut() {
+    if (isPaused) {
+      isPaused = false;
+      chrome.debugger.stepOut();
+    }
+  }
+  stepOver() {
+    if (isPaused) {
+      isPaused = false;
+      chrome.debugger.stepOver();
+    }
+  }
+
+  autoStepOver() => stepOver();
+
+  // TODO: do we need this?
+  stepOverAsyncSuspension() {}
```

Comment:
I don't think this would have an analog when debugging web apps (so, a no-op impl seems fine).